mlflow/gateway/providers/huggingface.py

```diff
@@ -0,0 +1,126 @@
+from typing import Any, Dict
+
+from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
+
+from mlflow.gateway.config import HuggingFaceTextGenerationInferenceConfig, RouteConfig
+from mlflow.gateway.providers.base import BaseProvider
+from mlflow.gateway.providers.utils import (
+    dict_contains_nested_path,
+    rename_payload_keys,
+    send_request,
+)
+from mlflow.gateway.schemas import chat, completions, embeddings
+
+
+class HFTextGenerationInferenceServerProvider(BaseProvider):
+    def __init__(self, config: RouteConfig) -> None:
+        super().__init__(config)
+        if config.model.config is None or not isinstance(
+            config.model.config, HuggingFaceTextGenerationInferenceConfig
+        ):
+            raise TypeError(f"Unexpected config type {config.model.config}")
+        self.huggingface_config: HuggingFaceTextGenerationInferenceConfig = config.model.config
+        self.headers = {"Content-Type": "application/json"}
+
+    async def _request(self, path: str, payload: Dict[str, Any]) -> Dict[str, Any]:
+        return await send_request(
+            headers=self.headers,
+            base_url=self.huggingface_config.hf_server_url,
+            path=path,
+            payload=payload,
+        )
+
+    async def chat(self, payload: chat.RequestPayload) -> chat.ResponsePayload:
+        raise HTTPException(
+            status_code=404,
+            detail="The chat route is not available for the Text Generation Inference provider.",
+        )
+
+    async def completions(self, payload: completions.RequestPayload) -> completions.ResponsePayload:
+        payload = jsonable_encoder(payload, exclude_none=True)
+        self.check_for_model_field(payload)
+        key_mapping = {
+            "prompt": "inputs",
+            "temperature": "parameters.temperature",
```

Comment:
This synthetic nesting by using "." seems a bit clunky. Can we just pop the prompt out and handle all other parameters in its own dictionary which is then added to the `payload` as a key `parameters` entry to conform to their swagger definitions? It will likely greatly simplify this implementation.

Reply:
Yes, we can do that as well. We had this in the [initial commit](https://github.com/mlflow/mlflow/pull/10072/commits/6cc752d58c69fae0f46dfe241a84049ed8f15569). We can revert it back.

Reply:
that would also work. i wanted to make this implementation similar to the other providers. but I can change the code so we don't have nested keys